tests/test_pipeline_files.py

```diff
@@ -526,6 +527,26 @@ def test_persist(self):
         self.assertIn(expected_policy, acl)
 
 
+    @defer.inlineCallbacks
+    def test_blob_path_consistency(self):
+        """Test to make sure that paths used to store files is the same as the one used to get
+        already uploaded files.
+        """
+        assert_gcs_environ()
+        import google.cloud.storage
+        with mock.patch('google.cloud.storage') as _ :
+            with mock.patch('scrapy.pipelines.files.time') as _ :
```

Comment:
Are these needed for the test to work as expected? I’m not used to patching without then using the patched object to define output and side effects.

Reply:
it may not be the best way of using monkeypatch but my idea was that I only care about making sure that `bucket.blob` & `bucket.get_blob` are called with the same `path`.
The problem with patching objects here is that we would need to patch a couple of them if we don't want to refactor GCSFilesStore.
I may be missing something but it's looks that we would need to patch is the following methods
- storage.Client 
- client.bucket
- bucket.test_iam_permissions
- bucket.get_blob
- bucket.blob
- blob.upload_from_string

if we would like to micmic that it works well but in practice this test is achieved by `test_persist`

The main advantage of using `mock.patch` here is that `storage` and all the above mentionned method will be [Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) objects which means that everytime you call an attritube or a method it will return a Mock object and will accept any arguments.
After that you can check which method has been called with which argument 
```
 store.bucket.blob.assert_called_with(expected_blob_path)
 store.bucket.get_blob.assert_called_with(expected_blob_path)
```
With those two lines I can make sure that `self.get_blob` from `stat_file` and `self.bucket.blob` from `persist_file` are called with the same and the right parameter.

Patching `time` is required because of `last_modified = time.mktime(blob.updated.timetuple())` but an alternative would have been to patch `blob.updated.timetuple` in order to return something that can be processed by `time.mktime`